core/common/src/main/java/alluxio/worker/block/io/LocalFileBlockReader.java

```diff
@@ -102,7 +102,9 @@ public ByteBuffer read(long offset, long length) throws IOException {
 
   @Override
   public int transferTo(ByteBuf buf) throws IOException {
-    return buf.writeBytes(mLocalFileChannel, buf.writableBytes());
+    int bytesReadFromCache = buf.writeBytes(mLocalFileChannel, buf.writableBytes());
+    MetricsSystem.counter(MetricKey.WORKER_BYTES_READ_CACHE.getName()).inc(bytesReadFromCache);
+    return bytesReadFromCache;
```

Comment:
`LocalFileBlockReader` is also used by `LocalFileDataReader` which is client side implementation that handles short circuit reads. If this worker specific metric is added here, then short circuit read will also cause worker cache read metrics to be updated.

The recommended place to inject this metric is in `alluxio.worker.block.TieredBlockStore#createBlockReader`. You can wrap the returned block reader object in some decorator class that's a subclass of `BlockReader`, and udpates the metrics in that decorator.